shared/bin/create-dashboard-rgw-user.sh

```diff
@@ -2,12 +2,30 @@
 
 set -e
 
+: ${CEPH_DEV_DOCKER_CONFIG_DIR:="$HOME/.ceph-dev-docker"}
+mkdir -p $CEPH_DEV_DOCKER_CONFIG_DIR
+
 #--------------
 # Configure RGW
 #--------------
 
 cd /ceph/build
 ./bin/radosgw-admin user create --uid=dev --display-name=Developer --system
 ./bin/ceph dashboard set-rgw-api-user-id dev
-./bin/ceph dashboard set-rgw-api-access-key `./bin/radosgw-admin user info --uid=dev | jq -r ".keys[0].access_key"`
-./bin/ceph dashboard set-rgw-api-secret-key `./bin/radosgw-admin user info --uid=dev | jq -r ".keys[0].secret_key"`
+
+RGW_ACCESS_KEY="${CEPH_DEV_DOCKER_CONFIG_DIR}/rgw_access_key"
+RGW_SECRET_KEY="${CEPH_DEV_DOCKER_CONFIG_DIR}/rgw_secret_key"
+./bin/radosgw-admin user info --uid=dev | jq -jr ".keys[0].access_key" > $RGW_ACCESS_KEY
+./bin/radosgw-admin user info --uid=dev | jq -jr ".keys[0].secret_key" > $RGW_SECRET_KEY
+chmod 600 $RGW_ACCESS_KEY
+chmod 600 $RGW_SECRET_KEY
+
+if ./bin/ceph --help 2>/dev/null | grep -q 'dashboard set-rgw-api-access-key.*<file>$'; then
```

Comment:
It comes out that the following code block does not work. Can you please replace it to:
```
if ! ./bin/ceph dashboard set-rgw-api-access-key -i $RGW_ACCESS_KEY; then
  ./bin/ceph dashboard set-rgw-api-access-key "$(cat $RGW_ACCESS_KEY)"
fi
if ! ./bin/ceph dashboard set-rgw-api-secret-key -i $RGW_SECRET_KEY; then
  ./bin/ceph dashboard set-rgw-api-secret-key "$(cat $RGW_SECRET_KEY)"
fi
```
The problem seems to be that the grep `'dashboard set-rgw-api-access-key.*<file>$'` does not work because the signature of the help texts has not been changed in https://github.com/ceph/ceph/pull/38284.

Reply:
OK, your solution looks much better :-)